src/OrchardCore.Modules/OrchardCore.Media.Azure/Middleware/MediaFileStoreMiddleware.cs

```diff
@@ -0,0 +1,212 @@
+        private readonly IMediaImageCache _mediaImageCache;
+        private readonly ICacheHash _cacheHash;
+        private readonly IRequestParser _requestParser;
+        private readonly IEnumerable<IImageProvider> _imageProviders;
+        private readonly FormatUtilities _formatUtilities;
+
+        private readonly PathString _assetsRequestPath;
+        private readonly string[] _allowedFileExtensions;
+
+        private readonly ImageSharpMiddlewareOptions _isOptions;
+
+        /// <summary>
+        /// The collection of known commands gathered from the processors.
+        /// </summary>
+        private readonly List<string> _isKnownCommands = new List<string>();
+
+        public MediaFileStoreMiddleware(
+            RequestDelegate next,
+            IMediaFileStore mediaFileStore,
+            IContentTypeProvider contentTypeProvider,
+            IMediaImageCache mediaImageCache,
+            ICacheHash cacheHash,
+            IRequestParser requestParser,
+            IEnumerable<IImageProvider> imageProviders,
+            IOptions<ImageSharpMiddlewareOptions> isOptions,
+            IEnumerable<IImageWebProcessor> processors,
+            IOptions<MediaOptions> mediaOptions,
+            ILogger<MediaFileStoreMiddleware> logger
+            )
+        {
+            _next = next;
+            _mediaFileStore = mediaFileStore;
+            _contentTypeProvider = contentTypeProvider;
+            _mediaImageCache = mediaImageCache;
+            _cacheHash = cacheHash;
+            _requestParser = requestParser;
+            _imageProviders = imageProviders;
+            _isOptions = isOptions.Value;
+
+            foreach (var processor in processors)
+            {
+                _isKnownCommands.AddRange(processor.Commands);
+            }
+
+            _formatUtilities = new FormatUtilities(_isOptions.Configuration);
+
+            _assetsRequestPath = mediaOptions.Value.AssetsRequestPath;
+            _allowedFileExtensions = mediaOptions.Value.AllowedFileExtensions;
+
+            Logger = logger;
+        }
+
+        public ILogger Logger { get; }
+
+        /// <summary>
+        /// Processes a request to determine if it matches a known file, and if so, serves it.
+        /// </summary>
+        /// <param name="context"></param>
+        public async Task Invoke(HttpContext context)
+        {
+            //TODO for 3.0 and endpoint routing, validate it is not an endpoint
+
+            if (!HttpMethods.IsGet(context.Request.Method) && !HttpMethods.IsHead(context.Request.Method))
+            {
+                await _next(context);
+                return;
+            }
+
+            var validatePath = context.Request.Path.StartsWithSegments(_assetsRequestPath, StringComparison.OrdinalIgnoreCase, out var subPath);
+            if (!validatePath)
+            {
+                Logger.LogDebug("Request path {0} does not match the assets request path {1}", subPath, _assetsRequestPath);
```

Comment:
This is not how log messages work. Log messages must be constant or they will allocate. That's why a different syntax is available for them.

```suggestion
                Logger.LogDebug("Request path {Path} does not match the assets request path {RequestPath}", subPath, _assetsRequestPath);
```

Reply:
Done